tf/modules/monitoring/signup-throttling-alert.tf

```
@@ -0,0 +1,39 @@
+resource "google_logging_metric" "api_signup_throttling_applied" {
+  name        = "${var.cluster_name}-api_signup_throttling_applied"
+  description = "A log based metric that detects when api was throttling sign up attempts"
+
+  # Filter for the secondary pod and look for readiness probe failures
+  filter = "resource.labels.cluster_name=\"${var.cluster_name}\" AND resource.labels.pod_name:\"api-app-\" AND resource.type=\"k8s_pod\" AND severity=WARNING AND jsonPayload.message:\"WARN_SIGNUP_THROTTLED\""
```

```suggestion
  filter = "resource.labels.cluster_name=\"${var.cluster_name}\" AND resource.labels.container_name:\"api-\" AND severity=ERROR AND resource.type=\"k8s_container\" AND textPayload:\"WARN_SIGNUP_THROTTLED\""
```

Right now this filter doesn't seem to capture the log lines emitted from the container. I'm guessing you hand wrote this rather than building it using the GCE UI? If you actually built it using the UI we might have a problems as perhaps the log objects aren't being made using consistent values. Otherwise this suggested change ought to work :crossed_fingers:

---

I hand wrote (adapting from the existing SQL alarms) this, but also tested it (against different log lines obviously) in the GCE UI which gave me results. It seems though that I missed that this catches K8s logs, not application level ones.

I am not sure about that ERROR severity though. Why is that needed?

---

Will the Laravel logger log to Stderr by default? I can confirm that the filter yields a line now.

---

https://github.com/wbstack/charts/blob/0206d86e811c491188f766f91f49f96555dc23b1/charts/api/templates/deployment-app-web.yaml#L169-L170